pennylane/io/qasm_interpreter.py

```diff
@@ -52,6 +64,156 @@
+COMPOUND_ASSIGNMENT_OPERATORS = [
+    "+=",
+    "-=",
+    "*=",
+    "/=",
+    "&=",
+    "|=",
+    "~=",
+    "^=",
+    "<<=",
+    ">>=",
+    "%=",
+    "**=",
+]
+COMPARISON_OPERATORS = [">", "<", ">=", "<="]
+BIT_SHIFT_OPERATORS = [">>", "<<"]
+
+NON_ASSIGNMENT_CLASSICAL_OPERATORS = (
+    EQUALITY_OPERATORS
+    + COMPARISON_OPERATORS
+    + BIT_SHIFT_OPERATORS
+    + [
+        PLUS,
+        DOUBLE_PLUS,
+        MINUS,
+        ASTERISK,
+        DOUBLE_ASTERISK,
+        SLASH,
+        PERCENT,
+        PIPE,
+        DOUBLE_PIPE,
+        AMPERSAND,
+        DOUBLE_AMPERSAND,
+        CARET,
+        AT,
+        TILDE,
+        EXCLAMATION_POINT,
+    ]
+)
+
+ASSIGNMENT_CLASSICAL_OPERATORS = [ARROW, EQUALS, COMPOUND_ASSIGNMENT_OPERATORS]
+
+
+class Context:
+    """Class with helper methods for managing, updating, checking context."""
+
+    def __init__(self, context):
+        if "vars" not in context:
+            context["vars"] = {}
+        if "aliases" not in context:
+            context["aliases"] = {}
+        self.context = context
+        if "wires" not in context:
+            context["wires"] = []
+
+    def update_var(
+        self,
+        value: any,
+        name: str,
+        node: QASMNode,
+    ):
+        """
+        Updates a variable, or raises if it is constant.
+        Args:
+            value (any): the value to set.
+            name (str): the name of the variable.
+            node (QASMNode): the QASMNode that corresponds to the update.
+        """
+        self.context["vars"][name]["val"] = value
+        if self.context["vars"][name]["constant"]:
+            raise ValueError(
+                f"Attempt to mutate a constant {name} on line {node.span.start_line} that was "
+                f"defined on line {self.context['vars'][name]['line']}"
+            )
+        self.context["vars"][name]["line"] = node.span.start_line
+
+    def require_wires(self, wires: list):
+        """
+        Simple helper that checks if we have wires in the current context.
+
+        Args:
+            wires (list): The wires that are required.
+
+        Raises:
+            NameError: If the context is missing a wire.
+        """
+        missing_wires = []
+        for wire in wires:
+            if wire not in self.context["wires"]:
+                missing_wires.append(wire)
+        if len(missing_wires) > 0:
+            raise NameError(
+                f"Attempt to reference wire(s): {missing_wires} that have not been declared in {self.context['name']}"
+            )
+
+    def __getattr__(self, name: str):
+        """
+        If the attribute is not found on the class, instead uses the attr name as an index
+        into the context dictionary, for easy access.
+        """
+        if name in self.context:
+            return self.context[name]
+        else:
+            raise NameError(f"No attribute {name} on Context and no {name} key found on context {self.context}")
```

Comment:
```suggestion
        raise NameError(f"No attribute {name} on Context and no {name} key found on context {self.context}")
```
Just a style thing, but you don't need `else`'s after a return statement. 

https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-else-return.html

Reply:
I am really trying to get used to this linting setting